### PR TITLE
fix(aggregation): use readPreference for aggregation

### PR DESCRIPTION
**Pull request by kvwalker**

### CODE REVIEW

File: lib/collection.js [hunk 1]
```diff
@@ -1742,6 +1742,7 @@ Collection.prototype.aggregate = function(pipeline, options, callback) {
   options = Object.assign({}, options);
   // Ensure we have the right read preference inheritance
   options.readPreference = resolveReadPreference(options, { db: this.s.db, collection: this });
+  command.readPreference = options.readPreference;
```
Comment (mbroadst) [inline lib/collection.js]: I think this is somewhat definitive proof that this error exists in `core` rather than `native`. We are correctly resolving the `readPreference` in the previous line, but you are able to identify that eventually the command generated in `core` is not decorated with the passed `readPreference`. Did you try to solve this there already?

File: test/functional/aggregation_tests.js [hunk 2]
```diff
@@ -1,7 +1,168 @@
...
+        collection.insert(docs, { w: 1 }, (err, result) => {
+          expect(result).to.exist;
+          expect(err).to.be.null;
+
+          const internalClientCursor = sinon.stub(client.topology.s.coreTopology, 'cursor');
+          const expectedReadPreference = new ReadPreference(ReadPreference.SECONDARY);
+
+          // Execute aggregate, notice the pipeline is expressed as an Array
+          collection.aggregate(
+            [
```
Comment (mbroadst) [inline test/functional/aggregation_tests.js]: do we need to use such a big pipeline for this, where we're testing just whether a readPreference makes it to the command? It seems like you could pass an empty array and save a lot of visual space

Comment (daprahamian) [reply test/functional/aggregation_tests.js]: Alternatively, you could do what the change stream tests do, and [make the pipeline a const outside the scope of the function](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/change_stream_tests.js#L15)

File: test/functional/aggregation_tests.js [hunk 3]
```diff
@@ -1,7 +1,162 @@
...
+                }
+              }
+            ],
+            { readPreference: new ReadPreference(ReadPreference.SECONDARY) },
+            (err, cursor) => {
+              expect(err).to.not.exist;
+              cursor.toArray(err => {
+                expect(err).to.be.null;
+                expect(cursor).to.not.be.null;
+                expect(internalClientCursor.getCall(0).args[2])
```
Comment (kvwalker) [inline test/functional/aggregation_tests.js]: It does look like the readPreference is being set correctly so it seems like NODE-1507 isn't actually a bug.

Comment (mbroadst) [reply test/functional/aggregation_tests.js]: I agree, can you please follow up on the ticket and try to elicit more information, letting them know what you've tried?
